### PR TITLE
Hide empty sessions from sidebar

### PR DESCRIPTION
**Pull request by justinchoo93**
## Summary
- Adds a "Hide empty sessions" setting (default: on) that filters out Claude Code sessions before any message is sent — no more hex ID placeholders cluttering the sidebar
- Uses `hasTitle` as the signal, which also catches sessions with system-only `firstUserMessage` (e.g. `<local-command-caveat>`)
- Adds urgent refresh path (300ms) in session indexer for titleless sessions so new sessions appear quickly once a title is set, rather than waiting for the normal 2-5s debounce/throttle cycle

## Test plan
- [x] Unit tests for `filterSessionItemsByVisibility` with `hideEmptySessions` on/off
- [x] Unit tests for system-only firstUserMessage edge case
- [x] Unit tests for urgent refresh timing in session indexer
- [x] Sort test helpers updated with `hasTitle` field
- [x] Settings default tests updated
- [x] Manual verification: empty sessions hidden, titled sessions appear quickly
- [x] All existing tests pass (1882 passing)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: server/coding-cli/session-indexer.ts [hunk 1]
```diff
@@ -430,20 +440,25 @@ export class CodingCliSessionIndexer {
...
-      if (this.throttleMs > 0 && fireElapsed < this.throttleMs) {
-        this.refreshTimer = setTimeout(() => {
-          this.refreshTimer = null
-          this.refresh().catch((err) => logger.warn({ err }, 'Refresh failed'))
-        }, this.throttleMs - fireElapsed)
-        return
+      // Skip this check for urgent refreshes (titleless session gained content).
+      if (!urgent) {
+        const fireElapsed = Date.now() - this.lastRefreshAt
+        if (this.throttleMs > 0 && fireElapsed < this.throttleMs) {
```
[the remark this answers is not in the document]
Comment (justinchoo93) [reply server/coding-cli/session-indexer.ts]: Good catch — fixed in 3e0c539. Urgent refreshes now use Math.min(URGENT_THROTTLE_MS, throttleMs) as the throttle floor instead of bypassing it entirely. This caps worst-case frequency at 1s for the default 5s throttle config and respects throttleMs: 0 (disabled) so urgent never becomes slower than normal. Added a test for the throttleMs: 0 edge case.
